src/Stan/Inspection/AntiPattern.hs

```diff
@@ -166,3 +172,23 @@ stan0206 = Inspection
     , inspectionSeverity = Performance
     , inspectionAnalysis = LazyField
     }
+
+-- | 'Inspection' — 'Foldable' methods on possibly error-prone structures @STAN-0207@.
+stan0207 :: Inspection
+stan0207 = mkAntiPatternInspection
+    (Id "STAN-0207")
+    "Foldable methods on possibly error-prone structures"
+    (FindAst combinedPattern)
+    & descriptionL .~ "Usage of Foldable methods on (,), Maybe, Either"
+    & solutionL .~
+        [ "Use more explicit functions with specific monomorphic types"
+        ]
+  where
+    combinedPattern :: PatternAst
+    combinedPattern = foldr1 PatternAstOr allPatterns
+
+    allPatterns :: NonEmpty PatternAst
+    allPatterns = do  -- Monad for NonEmpty
+        t <- foldableTypesPatterns
+        (method, mkType) <- foldableMethodsPatterns
+        pure $ PatternAstName method (mkType t)
```

Comment:
🤯 Clever one! But thank for the comment, without it, it would take me much longer to get!